.github/workflows/workflow.yml

```diff
@@ -14,30 +14,37 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ ubuntu-20.04, ubuntu-18.04, macos-11, windows-2019 ]
-        ruby: [ head, '3.0', 2.7, 2.6, 2.5, 2.4, jruby, truffleruby-head ]
+        os: [ ubuntu-20.04, macos-12, windows-2019 ]
+        ruby: [ head, 3.1, '3.0', 2.7, 2.4, jruby, truffleruby-head ]
```

Comment:
I don't think it's necessary to test all EOL rubies. Just keep testing 2.4 which is also the minimum based on Puma requirements